vpr/src/base/route_constraint.h

```diff
@@ -0,0 +1,65 @@
+#ifndef ROUTE_CONSTRAINT_H
+#define ROUTE_CONSTRAINT_H
+
+#include "vpr_types.h"
+
+/**
+ * @file
+ * @brief This file defines the RouteConstraint class.
```

Comment:
I think we should have a bigger comment on what this class is for, what it can currently do, what kind of code belongs here (so people know in the future if there is some new feature code that would best be put here), etc.